backend/battle/views.py

```diff
@@ -1,98 +1,57 @@
-#                                                              'message': error})
-
-
-# class RoundNewOponnent(View):
-#     form_class = RoundForm2
-#     initial = {'key': 'value'}
-#     template_name = 'battle/round_new2.html'
-#     battle_info = Battle.objects.latest('id')
-
-#     def get(self, request, *args, **kwargs):
-#             form_round2 = self.form_class(initial = self.initial)
-#             battle_get = self.battle_info
-#             return render(request, self.template_name, {'form_round2': form_round2,
-#                                                         'battle': battle_get})
-
-#     def post(self, request, *args, **kwargs):
-#             battle = self.battle_info
-#             form_round2 = self.form_class(request.POST, instance = battle)
-#             if form_round2.is_valid():
-#                 round_opponent = form_round2.save(commit=False)
-#                 data_pokemons = [
-#                     round_opponent.pk1_opponent,
-#                     round_opponent.pk2_opponent,
-#                     round_opponent.pk3_opponent
-#                 ]
-#                 info = get_total_point_pokemon(data_pokemons)
-#                 current_id = battle.id
-#                 if info == True:
-#                     pokemons = [round_opponent.pk1_opponent,
-#                                 round_opponent.pk2_opponent, round_opponent.pk3_opponent]
-#                     result = battleRunning(current_id, pokemons)
-#                     round_opponent.winner = result
-#                     round_opponent.save()
-#                     #result_battle()
-
-#                     return HttpResponseRedirect('/')
-#                 else:
-#                     error = message_error()
-#                     return render(request, self.template_name,
-#                                 {'battle': battle_info, 'message': error})
-
-
-# class Invite(View):
-#     template_name = 'battle/invite.html'
-
-#     def get(self, request):
-#         return render(request, self.template_name)
-
-
-# class Opponent(View):
-#     template_name = 'battle/opponent.html'
-
-#     def get(self, request):
-#         return render(request, self.template_name)
+
+class Invite(TemplateView):
+    template_name = 'battle/invite.html'
+
+
+class BattleView(CreateView):
+    model = Battle
+    template_name = 'battle/battle_form.html'
+    form_class = BattleForm
+
+    def form_valid(self, form):
+        form.instance.creator = self.request.user
+        form.save()
+        return HttpResponseRedirect(reverse_lazy("team_create", args=(form.instance.id, 1, )))
+
+
+class TeamView(CreateView):
+    model = Team
+    template_name = "battle/pokemon_form.html"
+    form_class = TeamForm
+    success_url = reverse_lazy("invite")
+
+    def get_initial(self):
+        super(TeamView, self).get_initial()
+        self.initial = {"battle": self.kwargs['pk'], "user": self.kwargs['user']}
```

Comment:
The initial fields should be the same ones as the form. In this case, `trainer` and `battle`.

Reply:
Also, instead of using the `self.kwargs['pk']` you can pass the battle object.

Reply:
thanks :D